src/js/jsx/shared/ColorInput.jsx

```diff
@@ -358,10 +358,36 @@ define(function (require, exports, module) {
                 }
             }
 
-            // swatch
-            swatchClassSet = classnames(swatchClassProps);
+            return {
+                value: value,
+                label: label,
+                color: color,
+                colorTiny: colorTiny
+            };
+        },
+
+        render: function () {
+            var colorInfo = this._extractColorInfo(),
+                value = colorInfo.value,
+                label = colorInfo.label,
+                color = colorInfo.color,
+                colorTiny = colorInfo.colorTiny;
+
+            var swatchClassProps = {
+                    "color-input": true
+                };
+
+            // setup text and swatch based on the mixed-ness of the inputs
+            if (value) {
+                if (typeof value === "string") {
+                    swatchClassProps["color-input__invalid-color"] = true;
```

Comment:
This is just a nit, but it seems like it would be even cleaner to just derive these two booleans (invalid, mixed) in `_extractColorInfo` since it is already doing similar logic, and then assign them to a single clean `swatchClassProps` here.